test/unit/data_hygiene/organisation_reslugger_test.rb

```diff
@@ -0,0 +1,103 @@
+require 'test_helper'
+require "gds_api/test_helpers/publishing_api"
+
+module OrganisationResluggerTest
+  module SharedTests
+    include GdsApi::TestHelpers::PublishingApi
+    extend ActiveSupport::Testing::Declarative
+
+    def self.included(klass)
+      klass.use_transactional_fixtures = false
+    end
+
+    def setup
+      DatabaseCleaner.clean_with :truncation
+      @organisation = create_organisation
+      @reslugger = DataHygiene::OrganisationReslugger.new(@organisation, 'corrected-slug')
+    end
+
+    def teardown
+      WebMock.reset!
+      DatabaseCleaner.clean_with :truncation
+    end
+
+    test "re-slugs the organisation" do
+      @reslugger.run!
+      assert_equal 'corrected-slug', @organisation.slug
+    end
+
+    test "publishes to Publishing API with the new slug and redirects the old" do
+      content_item = PublishingApiPresenters.presenter_for(@organisation).as_json
+      old_base_path = @organisation.search_link
+      new_base_path = "#{base_path}/corrected-slug"
+      content_item[:routes][0][:path] = new_base_path
+
+      redirects = [
+        { path: old_base_path, type: "exact", destination: new_base_path },
+      ]
+      redirects << { path: (old_base_path + ".atom"),
+                     type: "exact",
+                     destination: (new_base_path + ".atom") } if @organisation.is_a? Organisation
```

Comment:
This is an awfully long (multi)line to have a conditional hidden at the end. Could we move it to wrap the code instead?